client/src/main/java/io/pravega/client/stream/impl/Controller.java

```diff
@@ -219,5 +220,18 @@
      * @return Pravega node URI.
      */
     CompletableFuture<PravegaNodeUri> getEndpointForSegment(final String qualifiedSegmentName);
+    
+    /**
+     * Returns the total number of bytes from the provided positions until the stream.
+     * TODO: Make this API asynchronous.
+     * 
+     * @param stream The stream being queried.  
+     * @param checkpoint A set of segment/offset pairs for a single stream that represent a
+     *            consistent position in the stream. (IE: Segment 1 and 2 will not both appear in
+     *            the set if 2 succeeds 1, and if 0 appears and is responsible for keyspace 0-0.5
+     *            then other segments covering the range 0.5-1.0 will also be included.)
+     * @return The total number of bytes beyond the provided positions.
+     */
+    long getRemainingBytes(Stream stream, final Checkpoint checkpoint);
```

Comment:
Yes. I get that it is a bit odd. This is an internal interface that happens to need the exact same data structure with the same constraints as one for a public interface. The data is obviously used in different contexts and different ways, so I could invent a new class that does the same thing, but I decided as this was an internal interface I could use the same one. If you prefer I could create a new class called something like "cutpoint" that Checkpoint contains, and is used directly here.